File: src/containers/banners.ts
import { GachaBanner } from '../gacha/banner';

export class BannerContainer {
  private banners: GachaBanner[] = [];

  constructor() {
    this.bootstrap();
  }

  async bootstrap() {
    await this.reload();
  }

  async reload(): Promise<void> {
    this.banners = await GachaBanner.all();
  }

  addBanner(banner: GachaBanner): void {
    this.banners.push(banner);
  }

  all(): GachaBanner[] {
    return this.banners;
  }

  getBanner(id: string): GachaBanner | null {
    return this.banners.find((banner) => banner.id === id) || null;
  }
}

export const bannerContainer = new BannerContainer();

File: src/containers/banners.rs
use crate::gacha::banner::GachaBanner;

#[derive(Debug, Default)]
pub struct BannerContainer {
    banners: Vec<GachaBanner>,
}

impl BannerContainer {
    pub fn new() -> Self {
        Self::default()
    }

    pub fn reload(&mut self, banners: Vec<GachaBanner>) {
        self.banners = banners;
    }

    pub fn all(&self) -> &[GachaBanner] {
        &self.banners
    }

    pub fn get_banner(&self, id: &str) -> Option<&GachaBanner> {
        self.banners.iter().find(|b| b.id == id)
    }
}
